File: scala/fib.scala

object Fib {
  def main(args: Array[String]): Unit = {
    for (x <- fib(1, 1).take(args(0).toInt)) {
      println(x)
    }
  }

  def fib(x: Int, y: Int): LazyList[Int] = {
    return x #:: fib(y, x + y)
  }
}

File: kotlin/fib.kt



fun main(args: Array<String>) {
  for ( x in fib().take(args[0].toInt()) ) {
    println(x)
  }
}

fun fib(): Sequence<Int> {
  return generateSequence(Pair(1, 1)) { Pair(it.second, it.first + it.second) }
    .map { it.first }
}
